server/test/services/applicant/ApplicantServiceTest.java

```diff
@@ -1843,6 +1843,9 @@ public void relevantProgramsForApplicant_setsEligibilityOnMultipleApps() {
         .containsExactlyInAnyOrder(programForUnapplied.id, programDefinition.id());
     assertThat(result.unapplied().stream().map(ApplicantProgramData::isProgramMaybeEligible))
         .containsExactlyInAnyOrder(Optional.of(true), Optional.empty());
+    assertThat(
+            result.unappliedAndPotentiallyEligible().stream().map(ApplicantProgramData::programId))
```

Comment:
question (blocking): This test already seems a bit overloaded by testing many different things. Should we split out the tests for this new `unappliedAndPotentiallyEligible` function so that they're more isolated and can test a variety of behaviors?

Reply:
I moved this into it's own test that is also testing eligibility.

Reply:
Thanks for adding its own test!